packages/twenty-front/src/modules/object-record/record-group/states/selectors/visibleRecordGroupIdsComponentFamilySelector.ts

```diff
@@ -0,0 +1,84 @@
+import { recordGroupDefinitionFamilyState } from '@/object-record/record-group/states/recordGroupDefinitionFamilyState';
+import { recordGroupIdsComponentState } from '@/object-record/record-group/states/recordGroupIdsComponentState';
+import { RecordGroupDefinition } from '@/object-record/record-group/types/RecordGroupDefinition';
+import { RecordGroupSort } from '@/object-record/record-group/types/RecordGroupSort';
+import { recordGroupSortedInsert } from '@/object-record/record-group/utils/recordGroupSortedInsert';
+import { recordIndexRecordGroupHideComponentFamilyState } from '@/object-record/record-index/states/recordIndexRecordGroupHideComponentFamilyState';
+import { recordIndexRecordGroupSortComponentState } from '@/object-record/record-index/states/recordIndexRecordGroupSortComponentState';
+import { recordIndexRecordIdsByGroupComponentFamilyState } from '@/object-record/record-index/states/recordIndexRecordIdsByGroupComponentFamilyState';
+import { createComponentFamilySelectorV2 } from '@/ui/utilities/state/component-state/utils/createComponentFamilySelectorV2';
+
+import { ViewComponentInstanceContext } from '@/views/states/contexts/ViewComponentInstanceContext';
+import { ViewType } from '@/views/types/ViewType';
+import { isDefined } from '~/utils/isDefined';
+
+export const visibleRecordGroupIdsComponentFamilySelector =
+  createComponentFamilySelectorV2<RecordGroupDefinition['id'][], ViewType>({
+    key: 'visibleRecordGroupIdsComponentFamilySelector',
+    componentInstanceContext: ViewComponentInstanceContext,
+    get:
+      ({ instanceId, familyKey }) =>
+      ({ get }) => {
+        const recordGroupSort = get(
+          recordIndexRecordGroupSortComponentState.atomFamily({
+            instanceId,
+          }),
+        );
+        const recordGroupIds = get(
+          recordGroupIdsComponentState.atomFamily({
+            instanceId,
+          }),
+        );
+        const hideEmptyRecordGroup = get(
+          recordIndexRecordGroupHideComponentFamilyState.atomFamily({
+            instanceId,
+            familyKey,
+          }),
+        );
+
+        const result: RecordGroupDefinition[] = [];
+
+        const comparator = (
+          a: RecordGroupDefinition,
+          b: RecordGroupDefinition,
+        ) => {
+          switch (recordGroupSort) {
+            case RecordGroupSort.Alphabetical:
+              return a.title.localeCompare(b.title);
+            case RecordGroupSort.ReverseAlphabetical:
```

Comment:
Out of curiosity, how do we enable ReverseAlphabetical group sort ? I could not find

Reply:
In the sort menu
<img width="213" alt="Screenshot 2024-12-31 at 3 00 59 PM" src="https://github.com/user-attachments/assets/02eb5fa8-7925-4c31-82ad-d43a76a9efd5" />